src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,100 @@
+import React, {useEffect, useMemo, useState} from "react";
+import * as adminActions from "../../../actions/adminActions";
+import "./AdminPortal.scss";
+import {connect, useDispatch, useSelector} from "react-redux";
+import {bindActionCreators} from "redux";
+
+import Grid from "@material-ui/core/Grid";
+import {Button, Typography} from "@material-ui/core";
+
+function PanelManager(props) {
+/**
+ * Fetches, configures, and displays a grid-structured panel for a given 
+ * set in the store.
+ * 
+ * 
+ * Example: 
+ * { TODO}
+ * 
+ * @prop {array} operations List of operation types (i.e ["READ", "UPDATE" ) => options are: READ, UPDATE, DELETE 
+ */
+    const dispatch = useDispatch();
+    const api = useMemo(
+        () => ({
+            ...bindActionCreators(adminActions, dispatch),
+        }),
+        [dispatch]
+    );
+
+    let [records, setRecords] = useState({});
+
+    const addDefaults = (fields) => props.fields.map((field) => {
+        //Add defaults field properties to the props.fields object
+        Object.keys(defaults).forEach((key) => {
+            if(field.hasOwnProperty(key)) {
+                //Do nothing, property is already set
+            } else {
+                field[key] = defaults[key];
+            }
+        });
+        return field;
+    });
+
+    const defaults = {
+        "editable": "false",
+        "align": "left"
+    };
+
+
+    const generateGrid = () => {
+        /**
+         * Generates header with fields
+         */
+        const fieldsWithDefaults = addDefaults(props.fields);
+        const headerElements = [];
+        const records = [];
+
+        for (const [index, value] of fieldsWithDefaults.entries()) {
+
+            headerElements.push(
+                <Grid item xs={value["col-width"]} md={value["col-width"]} >
+                    <Typography align={value["align"]} style={{color: 'white', fontWeight: '500'}}>
+                        {value["name"]}
+                    </Typography>
+                </Grid>
+            )
+        }
+        for (const [index, value] of props.categories.entries()) {
+            records.push(
+                <li>{value["name"]}</li>
+            )
+        };
+
+        return (
+            <Grid container>
+                {/* header */}
+                <Grid item xs={12}>
+                    <Grid container className={'accounts-table-heading'}>
+                    {headerElements}
+                    </Grid>
+                </Grid>
+
+                {/* Directory */}
+                <Grid>
```

Comment:
I haven't actually pulled/looked at this, but can you verify that the "Directory" is the list of items we're displaying? For example the red box highlighted here would be the directory:
![Screen Shot 2020-03-05 at 4 20 16 PM](https://user-images.githubusercontent.com/12959959/76037977-483e4a00-5efd-11ea-9693-a6129d2c90d9.png)
If this is it, is this the final version? If you're still working on making the update functionalities, then I'll just let you finish before I say anything, haha.

Reply:
Yes it is, but I need to style the records. The headings work but the records are currently just an <li>.

As for the edit function, it’s looking decent I just didn’t wanna confuse you because right now it’s too specific for the categories component, I need to rename stuff to make it more general and get it working for other field types. I’ll commit tonight

Reply:
sounds good! rock on, man :)